src/index.js

```diff
@@ -39,6 +46,55 @@ class Offline {
     this.exitCode = 0;
     this.provider = 'aws';
     this.start = this.start.bind(this);
+    this.coverageDataOutputPath = process.env.SLS_OFFLINE_COVERAGE_DATA_OUTPUT_PATH;
+    this.shouldOutputCoverage = typeof this.coverageDataOutputPath !== 'undefined';
+
+    if (this.shouldOutputCoverage) {
+      debugLog(
+        `Code coverage will be written to ${this.coverageDataOutputPath}`
+      );
+      /* eslint-disable new-cap */
+      this.coverageQueue = new pQueue({ concurrency: 1 });
+      /* eslint-enable new-cap */
+      this.outputCoverage = pDebounce(
+        coverageData =>
+          this.coverageQueue.add(async () => {
+            debugLog('_____ COVERAGE _____');
+
+            if (coverageData == null) {
+              console.warn('[offline] warning: was instructed to output coverage data, but no coverage data could be found after executing the lambda function handler. Most likely the handler code is not instrumented.');
+
+              return;
+            }
+
+            await writeFile(
+              this.coverageDataOutputPath,
+              JSON.stringify(coverageData),
+              'utf8',
+            );
+
+            console.log(
+              `Coverage data written to ${this.coverageDataOutputPath}`
+            );
+
+            if (typeof process.env.SLS_DEBUG !== 'undefined') {
+              const coverageMap = libCoverage.createCoverageMap(coverageData);
+              const summary = coverageMap
+                .files()
+                .reduce(
+                  (sum, f) => sum.merge(coverageMap.fileCoverageFor(f).toSummary()),
+                  libCoverage.createCoverageSummary(),
+                );
+              debugLog('Global coverage summary', summary);
+            }
+          }),
+        // TODO: perhaps make this configurable
```

Comment:
Thoughts? This value is the debounce duration.

Reply:
You could move it to a constant, to make clarify this magic number. I do not foresee an usecase for changing this value.